controller/eks-cluster-config-handler.go

```diff
@@ -1115,15 +1124,19 @@ func (h *Handler) updateUpstreamClusterState(upstreamSpec *eksv1.EKSClusterConfi
 		// in this case update is set right away because creating the
 		// nodegroup may not be immediate
 		if config.Status.Phase != eksConfigUpdatingPhase {
-			config = config.DeepCopy()
 			config.Status.Phase = eksConfigUpdatingPhase
-			var err error
 			config, err = h.eksCC.UpdateStatus(config)
 			if err != nil {
 				return config, err
 			}
 		}
-		ltVersion, err := createNodeGroup(config, ng, eksService, ec2Service, svc)
+		ltVersion, role, err := createNodeGroup(config, ng, eksService, ec2Service, svc)
+
+		// if the generated node role hasn't been set yet and it just got generated, save
+		// it to set on the Status
+		if generatedNodeRole == "" && role != "" {
```

Comment:
Just to confirm, the intention here is that the last node in the node group becomes to `config.Status.GeneratedNodeRole`? Would there be any significance to 2 node groups having different generated node roles?

Reply:
Left a clarifying comment here: https://github.com/rancher/eks-operator/pull/73/files/929ccff9443e3aa0f2e3468e71e6b5b7474759ce#r1025781996